File: dotnet/cafdemalihapa/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/PowerShellScripts/PowerShellScriptsDirectory.cs
using cafdemalihapa.Directories.Hosting.Feature.Automations.Commands;

namespace cafdemalihapa.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.PowerShellScripts
{
    public class PowerShellScriptsDirectory
    (
        ScriptsDirectory scriptsDirectory,
        Directories directories,
        CommandsDirectory commandsDirectory
    )
    {
        const string directoryNameInSourceCode = "powershell-scripts";

        public void CopyContentToDirectory()
        {
            var destinationDirectory = commandsDirectory.GetPath();
            var sourceDirectory = GetPath();
            directories.CopyContentOfSourceDirectoryToDestinationDirectory(sourceDirectory, destinationDirectory);
        }

        public string GetPath()
        {
            var scriptsDirectoryName = scriptsDirectory.GetPath();
            var powerShellDirectoryPathInSourceCode = Path.Combine(scriptsDirectoryName, directoryNameInSourceCode);
            return powerShellDirectoryPathInSourceCode;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Repository/Dotnet/Scripts/PowerShellScripts/PowerShellScriptsDirectory.fs
namespace cafdemwimu.console.Directories.Repository.Dotnet.Scripts.PowerShellScripts

open System.IO
open cafdemwimu.console.Directories
open cafdemwimu.console.Directories.Hosting.Feature.Automations.Commands
open cafdemwimu.console.Directories.Repository.Dotnet.Scripts

type PowerShellScriptsDirectory(scriptsDirectory: ScriptsDirectory, commandsDirectory: CommandsDirectory) =
    let directoryNameInSourceCode = "powershell-scripts"

    member this.CopyContentToDirectory() =
        let destinationDirectory = commandsDirectory.GetPath()
        let sourceDirectory = this.GetPath()
        DirectoryServices.CopyContentOfSourceDirectoryToDestinationDirectory(sourceDirectory, destinationDirectory)

    member _.GetPath() =
        let scriptsDirectoryName = scriptsDirectory.GetPath()
        let powerShellDirectoryPathInSourceCode = Path.Combine(scriptsDirectoryName, directoryNameInSourceCode)
        powerShellDirectoryPathInSourceCode
